DOC-5174: Added configuring users content (#1287)
corda/corda-docs-portal: content/en/platform/corda/5.0/deploying-operating/config-users/permission-system-entities.md

* `E-mail` is an email address for the user account used for communication of any important information
  affecting this account. Same email address may be used for multiple accounts.
* `Salt value` is a randomly generated value used when hashing the password. It only makes sense when password is
  assigned as well.
* `Hashed password` ia a salted and hashed representation of the password.
 This property can be set to `null` for SSO-only accounts.
* `Password expiry` is only applicable when password is assigned and when it specifies the timestamp after which the password
  is deemed expire and needs changing.
* `Parent Group` is an optional identifier of the group to which the user belongs. This property can be se to `null`. In such
  case, the user is assumed to belong to the "root" group. In the current model, the user can belong to just a
  single group. However, groups can be nested.

### Relationships

* `User` can be associated one-to-one with a `Group`.
* `User` can have multiple roles associated with it.
* `Change Audit` entries may have one-to-one references to acting `User`.
* `User` can have multiple `Properties` assigned to it.

> ### Properties

Reviewer: Since it is not currently possible to add, remove, modify properties, I would remove this section from the documentation.

Author: @vkolomeyko would it be possible for GA?

Reviewer: No, it will not be implemented for GA.

Author: removed

Properties here mean a set of `Key` and `Value` pairs associated with either `User` or `Group`.
Users of the RBAC database are able to retrieve these properties and make decisions depending on the context.

* `Key` is a string representing a value. Only one unique key is allowed for a given `User` or `Group`.
* `Value` is a string representing a value associated with a `Key`.

> ## Group

Author: @vkolomeyko will they be implemented for GA?

> ## Group

Reviewer: No, it will not be implemented for GA.

> ## Group

Author: removed

---
date: '2023-04-24'
title: "Permission System Entities"
menu:
  corda-5:
    parent: corda-5-cluster-config-users
    identifier: corda-5-cluster-permission-system-entities
    weight: 3000
section_menu: corda-5
---

> R3's role-based access control (RBAC) permissions system is made up of multiple entities.

Author: ```suggestion
Corda 5 role-based access control (RBAC) permissions system is made up of multiple entities.
```

> The R3's role-based access control (RBAC) permissions system is made up of multiple entities.

Reviewer 2: I'd suggest "Corda 5" rather than R3?

The following section describes those entities and relationships between them.

## User

This entity represents a human user or system user.

> ### Properties

Reviewer: ```suggestion
### Attributes
```
I would make such a rename to avoid confusion with `Property` entity that we have as part of RBAC entity model.

  the instance of an entity.

### Relationships

Each role can be associated with multiple `Permissions`. A given `Permission` can belong to multiple roles. Hence,
there is many-to-many relationship between `Role` and `Permission`.

## Permission

This entity represents an individual fine-grained permission.

### Properties

* `Virtual Node ID` is an optional identifier of the virtual node within physical node the permission applies to.
* `Permission type` defines whether this is an "allow" or "deny" type of permission. "Deny" permissions will always
  have advantage over "allow", no matter at which level (`User` or `Group`) they are granted.
* `Permission string` is a machine-parseable string representing individual permission. It can be an arbitrary string as
  long as authorization code could make a good use of it in the context where `User` permission to perform a certain
  operation is checked. For example, it can be in the form: `InvokeRpc:...`, or `StartFlow:...`, or `CreateUser:...`.

## Change Audit

This entity represents an audit log for every change to the permission data. This can be introduction of a new `User`,

> assignment `User` to a `Group`, creation of a `Role`, and so on.

Reviewer 3: ```suggestion
assignment of a `User` to a `Group`, creation of a `Role`, and so on.
```
